custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -308,6 +308,22 @@ window.angular.module('icdsApp').factory('baseControllersService', function() {
                 vm.mobilePopupLocation = location;
                 vm.mobilePopupData = data;
             };
+            vm.drilldownToLocationWithName = function (locationName) {
+                // todo: this is heavily copied from map-or-sector-view's handling of the map click event
+                // but there's not a great place to share the code since they're managed by separate controllers
+                var currentLocationId = $location.search().location_id;
+                locationsService.getLocationByNameAndParent(locationName, currentLocationId).then(function (locations) {
+                    var location = locations[0];
+                    $location.search('location_name', location.name);
+                    $location.search('location_id', location.location_id);
+
+                    storageService.setKey('search', $location.search());
+                    if (location.location_type_name === 'awc') {
+                        $location.path('awc_reports');
```

Comment:
does this preserve the report you are looking at (or is it too early to even be asking this question)? For the web, when you drill down to the awc it shows you the awc report that most closely corresponds to the other report you were looking at.

Reply:
Good catch. It did not, but thanks to https://github.com/dimagi/commcare-hq/pull/26227 it was easy to add in https://github.com/dimagi/commcare-hq/pull/26223/commits/5b69ceeb3bcb8511f857145a66db385ec1580360